### ISSUE TITLE
With Cocos2dx V3.9, Admob Crash in Android

### ISSUE THREAD

**Issue by ghost** [SonarSystems/Cocos-Helper#36]
Hi, when I use this helper, the following error exist:

java.lang.NullPointerException
sonar.systems.framework.SonarFrameworkFunctions.ShowBannerAd(SonarFrameworkFunctions.java:721)

My app crash when try to load the Admob. Could you help to solve? Thanks

**Comment by SonarSystems** [comment]
Have you setup Cocos Helper and AdMob?

**Comment by OscarLeif** [comment]
That sounds very strange did you watch the videos from sonar systems ? Or that happens with that method only ?

**Comment by ghost** [comment]
Yes, I've watch the video serval time, and even recreate the cocos hello world project and integrate the Helper. Also failed.

**Comment by ghost** [comment]
Is there any change made by the new version of cocos2dx that will impact the JNIHelper works right? Can you do some investigate on that? Thanks

**Comment by OscarLeif** [comment]
I'm gonna check if the new version of google play services is the problem

**Comment by OscarLeif** [comment]
Finally I have new for you @alexshihu  I make it work but It seems to work fine. I made no changes to the code. No crash or something like that, maybe you have a bad configuration in your project, ok lets start what maybe could be wrong.
- Update your google play services library I test it with version 28
- Clean your project and then Build
- Remember import the google play services jar file, of course remember to mark it in the tab "Order and Export"
- Please compile Cocos2d-x game project and Google play services project with Android API 19, I'm not testing it with Android 5 + but it should work.
- Of course remember you ads units and **PERMISSIONS** maybe it doesn't have internet permissions
- Did you change the AndroidManifest to load GooglePlayServices and AdMob ?

I test it in a motorola moto G first gen. Also test fullscreen intersitial.
It could be just bad configuration.
